### PR TITLE
Fixed cleaning operation storage when using `async` mode and forking

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/disk_queue.py [hunk 1]
```diff
@@ -167,14 +167,15 @@ def flush(self):
...
     def close(self):
-        """
-        Close and remove underlying files if queue is empty
-        """
         self._reader.close()
         self._writer.close()
         self._last_ack_file.close()
         self._last_put_file.close()
 
+    def cleanup(self) -> None:
```
Comment (PatrykGala) [inline src/neptune/internal/disk_queue.py:175]: `cleanup_if_empty`

File: src/neptune/internal/operation_processors/async_operation_processor.py [hunk 2]
```diff
@@ -204,7 +204,12 @@ def stop(self, seconds: Optional[float] = None):
...
-        self._queue.close()
+
+        # Close resources
+        self.close()
+
+        # Remove local files
+        self._queue.cleanup()
 
     def close(self):
         self._queue.close()
```
Comment (PatrykGala) [inline src/neptune/internal/operation_processors/async_operation_processor.py:215]: Is the function `stop` always called before `close`? Prior to these changes, the close function used to remove files.

Comment (Raalsky) [reply src/neptune/internal/operation_processors/async_operation_processor.py:215]: > Prior to these changes, the close function used to remove files.
And `close` was called in the fork child process sometimes causing directories removal of a root process disk queue etc.

File: src/neptune/internal/disk_queue.py (hunk 1 above)
Comment (Raalsky) [reply src/neptune/internal/disk_queue.py]: Sure

**Review by PatrykGala — APPROVED** [review]

**Pull request merged**